app/routers/user_exercise.py

```diff
@@ -0,0 +1,55 @@
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from datetime import datetime
+from app.database.models import Base, User, UserExercise
+
+
+def save(session: Session, instance: Base) -> bool:
+    """Commits an instance to the db.
+    source: app.database.models.Base"""
+
+    if issubclass(instance.__class__, Base):
+        session.add(instance)
+        session.commit()
+        return True
+    return False
+
+
+def create_user_exercise(session: Session, user: User) -> UserExercise:
+    """
+    Create and save new user exercise
+    """
+    if not does_user_exercise_exist(session=session, user_id=user.id):
+        user_exercise = UserExercise(
+            user_id=user.id,
+            start_date=datetime.now()
+            )
+        save(session=session, instance=user_exercise)
+    else:
+        user_exercise = update_user_exercise(session=session, user=user)
+    return user_exercise
+
+
+def does_user_exercise_exist(session: Session, user_id: int) -> bool:
+    """
+      Check if a user exercise for user id exists.
+    """
+    return get_user_exercise(session=session, user_id=user_id)
+
+
+def get_user_exercise(session: Session, **param) -> list:
+    """Returns user exercise filter by param."""
+    try:
+        user_exercise = list(session.query(UserExercise).filter_by(**param))
+    except SQLAlchemyError:
+        return []
+    else:
+        return user_exercise
+
+
+def update_user_exercise(session: Session, user: User) -> UserExercise:
+    user_ex = session.query(UserExercise).filter_by(user_id=user.id).first()
+    # Update database
+    user_ex.start_date = datetime.now()
+    session.commit()
+    return user_ex
```

Comment:
Did you took the pictures inside `images/exercises`? If not - what is the source? What is the license?

Reply:
All pictures are came from "Lose weight in 30 days" application , Its free application

https://www.microsoft.com/en-us/p/lose-weight-in-30-days/9nx88f64d54n?activetab=pivot:overviewtab

Reply:
The fact that it's free application doesn't imply you can use the images from the app.

You should check what is the license of the images. You can email them or use other, free for use images.
Using images without permission can lead to hard copyright issues.

Reply:
I send an email to microsoft support , waiting for their response

Reply:
Hi Yam
I still wating to response from Microsoft support ,
Meanwhile, I write the contact by my self and I remove all pictures, Because of copyright,
I convert exercise files from png to html file.
I create new path for html files  and i change the config file , so you can approve my pull request.

I Microsoft support will return my and alowed my to use their images and screen shot , I just chage the path in the config file.

Hopefully now you can approve my pull request

Thanks

Oron